### PR TITLE
Watchdog to call probe-point when SocketPolls are slow.

### PR DESCRIPTION
**Pull request by mmeeks**
Change-Id: Id58efab0c0fee88a3da094765eb6b7641aec6e02

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Review by vmiklos — APPROVED** [review]
Looks good to me in general. Some comments:

- if it turns out that the 100ms limit depends on machine speed, then perhaps COOL_WATCHDOG could have an int value and that would specify what should be the custom limit, instead of the 100ms default
- nothing jumps out as I code-read this, but probably good to wait for CI to finish before merging.

**Review by caolanm — APPROVED** [review]

**Pull request merged**
